restricted_site_access.php

```diff
@@ -1529,36 +1529,85 @@ public static function ip_in_range( $ip, $range ) {
 	 * @return string
 	 */
 	public static function get_client_ip_address() {
-		$ip      = '';
-		$headers = array(
-			'HTTP_CF_CONNECTING_IP',
-			'HTTP_CLIENT_IP',
-			'HTTP_X_FORWARDED_FOR',
-			'HTTP_X_FORWARDED',
-			'HTTP_X_CLUSTER_CLIENT_IP',
-			'HTTP_FORWARDED_FOR',
-			'HTTP_FORWARDED',
-			'REMOTE_ADDR',
-		);
-		foreach ( $headers as $key ) {
-
-			if ( ! isset( $_SERVER[ $key ] ) ) {
+		$ip                    = '';
+		$remote_addr_header_ip = isset( $_SERVER['REMOTE_ADDR'] ) ? sanitize_text_field( wp_unslash( $_SERVER['REMOTE_ADDR'] ) ) : false;
+	
+		/**
+		 * Assume empty REMOTE_ADDR as unreliable.
+		 */
+		if ( false === $remote_addr_header_ip ) {
+			return '';
+		}
+	
+		/**
+		 * Accepts the string 'REMOTE_ADDR' or array of trusted proxies.
+		 * It is advisable to pass 'REMOTE_ADDR' if your proxy server doesn't have
+		 * a static IP.
```

Comment:
If my assumption is correct that this should be utilized to add trusted IP addresses, I think we need to change the comment here to make that clear. Right now, this is confusing to me

Reply:
> If my assumption is correct that this should be utilized to add trusted IP addresses

Correct. If the array is not set, then the default string `REMOTE_ADDR` is used. The situation where you won't set this array is when the proxy has dynamic IPs. in that case we trust any value that is set in `REMOTE_ADDR`